### PR TITLE
feat: add static serving components and Chicken Flavored Miss Dash seasoning

### PR DESCRIPTION
**Pull request by mbuchoff**
## Summary
- Add StaticFoodServing wrapper to prevent scaling of specific components
- Implement FromComponentsWithStatic factory method for mixed component types
- Add Chicken Flavored Miss Dash seasoning with zero macros
- Update Seitan composite to include static seasoning component

### StaticFoodServing Class
- Wrapper around FoodServing that prevents scaling when multiplied
- Key behavior: `operator *` returns unchanged instance
- Preserves original serving through delegation pattern

### Enhanced CompositeFoodServing
- New `FromComponentsWithStatic` factory method
- Runtime type checking for proper scaling behavior
- Supports mixed scalable and static components in single composite

### Practical Application
- Seitan now includes 1 tbsp Chicken Flavored Miss Dash as static component
- When Seitan serving is doubled, seasoning stays at 1 tbsp (realistic cooking behavior)
- Scalable components (nutritional yeast, gluten) scale normally

## Technical Implementation
- **Challenge**: C# operator dispatch is compile-time, not runtime
- **Solution**: Runtime type checking in CompositeFoodServing display methods
- **Pattern**: Polymorphic method dispatch with `is` type checking

## Test Coverage
- 10 comprehensive test cases in StaticComponentTests.cs
- Tests cover basic behavior, mixed scaling, string output, nutrition calculation
- Edge cases: complex nesting, empty components, factory methods

## Benefits
- **Realistic cooking behavior** - seasonings don't scale linearly
- **Flexible architecture** - any component can be made static
- **Clean API** - explicit factory method shows intent
- **Backward compatible** - existing functionality unchanged

## Use Cases
Perfect for:
- Salt, spices, seasonings
- Garnishes and toppings  
- Small amounts of cooking oils
- Supplements or vitamins

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: Executable/CompositeFoodServing.cs [hunk 1]
```diff
@@ -72,10 +113,18 @@ public override IEnumerable<FoodServing> GetComponentsForDisplay()
...
-            yield return component;
+            // Check if this is a static component - if so, don't scale it
+            if (component is StaticFoodServing staticComponent)
+            {
+                yield return staticComponent; // Return unchanged
+            }
+            else
+            {
+                yield return component * scale; // Scale regular components
+            }
```
Comment (mbuchoff) [inline Executable/CompositeFoodServing.cs:127]: Same as with previous comment.
